services/slam/fake/slam.go

```diff
@@ -66,6 +100,13 @@ type SLAM struct {
 	logger    golog.Logger
 }
 
+func (slamSvc *SLAM) count() int {
+	if slamSvc.dataCount < 0 {
```

Comment:
What would be the effect of deferring the call to `slamSvc.incrementCount() `

Reply:
The effect will be that the count won't be incremented until the map call responds, which means that if you have 2 in-flight GetMap requests, both will get the same data. Also, GetPosition requests won't know whether they will get the data before or after the GetMap call completes b/c the client doesn't control how long it takes the image or internal state data takes to load from the disk.

Subjectively I found that the deferred increment resulted in more confusing behavior as a user, given that it affects every single call, rather than what I've implemented here as it only affects the very first call to GetMap.